.cicdtemplate/.github/workflows/deploy_production_to_firebase_app_distribution.yml

```diff
@@ -0,0 +1,68 @@
+name: Deploy staging and production to Firebase App Distribution
+
+on:
+  # Trigger the workflow on push action in main branch.
+  # So it will trigger when the PR of the feature branch was merged.
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy_staging_and_production_to_firebase_app_distribution:
+    name: Deploy staging and production to Firebase App Distribution
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Set up timezone
+        uses: zcong1993/setup-timezone@master
+        with:
+          timezone: Asia/Bangkok
+
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Run Detekt
+        run: ./gradlew detekt
+
+      - name: Archive Detekt reports
+        uses: actions/upload-artifact@v3
+        with:
+          name: DetektReports
+          path: build/reports/detekt/
+
+      - name: Run unit tests with Kover
+        run: ./gradlew koverHtmlReport
+
+      - name: Archive code coverage reports
+        uses: actions/upload-artifact@v3
+        with:
+          name: CodeCoverageReports
+          path: app/build/reports/kover/
+
```

Comment:
IMO, we don't need these code analyzing & testing steps on CD workflows. Let's save our billing resources 🙏 Same for staging workflow